ch_backup/clickhouse/disks.py

```diff
@@ -87,63 +98,66 @@ def __exit__(self, exc_type, *args, **kwargs):
             logging.debug(f"Removing tmp disk {disk.name}")
             try:
                 os.remove(_get_config_path(self._config_dir, disk.name))
+                self._disks.pop(disk.name)
                 return True
             except FileNotFoundError:
                 pass
 
+    def _render_disks_config(self, path, disks):
+        with open(path, "w", encoding="utf-8") as f:
+            xmltodict.unparse(
+                {
+                    "yandex": {
+                        "storage_configuration": {"disks": disks},
+                    }
+                },
+                f,
+                pretty=True,
+            )
+
     def _create_temporary_disk(
         self,
         backup_meta: BackupMetadata,
-        disk: Disk,
+        disk_name: str,
         source_bucket: str,
         source_path: str,
         source_endpoint: str,
     ) -> None:
-        tmp_disk_name = _get_tmp_disk_name(disk.name)
+        tmp_disk_name = _get_tmp_disk_name(disk_name)
         logging.debug(f"Creating tmp disk {tmp_disk_name}")
-        disk_config = self._ch_config.config["storage_configuration"]["disks"][
-            disk.name
-        ]
+        disk_config = copy.copy(
+            self._ch_config.config["storage_configuration"]["disks"][disk_name]
+        )
 
         endpoint = urlparse(disk_config["endpoint"])
         endpoint_netloc = source_endpoint or endpoint.netloc
         disk_config["endpoint"] = os.path.join(
             f"{endpoint.scheme}://{endpoint_netloc}", source_bucket, source_path, ""
         )
-        with open(
-            _get_config_path(self._config_dir, tmp_disk_name), "w", encoding="utf-8"
-        ) as f:
-            xmltodict.unparse(
-                {
-                    "yandex": {
-                        "storage_configuration": {
-                            "disks": {
-                                tmp_disk_name: disk_config,
-                            },
-                        },
-                    }
-                },
-                f,
-                pretty=True,
-            )
+        self._render_disks_config(
+            _get_config_path(self._config_dir, tmp_disk_name),
+            {tmp_disk_name: disk_config},
+        )
```

Comment:
Do we still need this file? Isn't tmp disk needed only for clickhouse-disks?

Reply:
It is a config for  clickhouse-server, we need it to do not create tmp_disk manually (like data_path and downloading metadata)